boards/adafruit-grand-central-m4-express/include/periph_conf.h

```diff
@@ -104,7 +104,7 @@ static const adc_conf_chan_t adc_channels[] = {
  */
 static const i2c_conf_t i2c_config[] = {
     {
-        .dev      = &(SERCOM3->I2CM),
+        .sercom   = 3,
```

Comment:
We already have `sercom_id()`, that would be less invasive

Reply:
Whats the verdict? Use `uint8_t sercom` and touch all boards for better efficiency, or stick with `void *dev` for backward compatibility with out-of-tree boards?

Reply:
@benpicco ping? I'd like to push this forward a bit. If you prefer not touching the `periph_conf.h`, I'm fine with that and can update the PR accordingly.

Reply:
@kfessel convinced me that keeping the `periph_conf.h` format is easier.